libs/api/domains/rsk-company-info/src/lib/rsk-company-info-module.ts

```diff
@@ -0,0 +1,37 @@
+import { DynamicModule, Module } from '@nestjs/common'
+import {
+  RskCompanyInfoServiceOptions,
+  CompanyApi,
+  Configuration as RestConfiguration,
+} from '@island.is/clients/rsk-company-info'
+import { RskCompanyInfoResolver } from './graphql/rsk-company-info.resolver'
+import { RskCompanyInfoService } from './graphql/rsk-company-info.service'
+
+@Module({})
+export class RskCompanyInfoModule {
+  static register(config: RskCompanyInfoServiceOptions): DynamicModule {
+    const { xRoadBaseUrl, xRoadProviderId } = config
+    const RSK_COMPANY_BASE_PATH = `${xRoadBaseUrl}/r1/${xRoadProviderId}/`
+    return {
+      module: RskCompanyInfoModule,
+      providers: [
+        RskCompanyInfoResolver,
+        RskCompanyInfoService,
+        {
+          provide: CompanyApi,
+          useFactory: () =>
+            new CompanyApi(
+              new RestConfiguration({
```

Comment:
This stuff can be done in a RskCompanyInfoModule which is part of the RskCompanyInfo client library. Check out VMST as an example, [client module](https://github.com/island-is/island.is/blob/a737ca6b15e902fe9d1e214d87d3e585774fc7e7/libs/clients/vmst/src/lib/vmst.module.ts#L24) vs [domain](https://github.com/island-is/island.is/blob/a737ca6b15e902fe9d1e214d87d3e585774fc7e7/libs/api/domains/directorate-of-labour/src/lib/directorate-of-labour.module.ts#L30-L41).